<a href="https://colab.research.google.com/github/isra9731/projects/blob/main/Exploratory_Data_Analysis_Retail.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##**Exploratory Data Analysis - Retail**



*    Perform ‘Exploratory Data Analysis’ on dataset  ‘SampleSuperstore’
*    As a business manager, try to find out the weak areas where you can work to make more profit.
*    What all business problems you can derive by exploring the data?


DATASET Used : https://bit.ly/3i4rbWl

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import os
%matplotlib inline

In [2]:
data=pd.read_csv("SampleSuperstore.csv")
data

Ship Mode    Segment        Country  ... Quantity Discount    Profit
0       Second Class   Consumer  United States  ...        2     0.00   41.9136
1       Second Class   Consumer  United States  ...        3     0.00  219.5820
2       Second Class  Corporate  United States  ...        2     0.00    6.8714
3     Standard Class   Consumer  United States  ...        5     0.45 -383.0310
4     Standard Class   Consumer  United States  ...        2     0.20    2.5164
...              ...        ...            ...  ...      ...      ...       ...
9989    Second Class   Consumer  United States  ...        3     0.20    4.1028
9990  Standard Class   Consumer  United States  ...        2     0.00   15.6332
9991  Standard Class   Consumer  United States  ...        2     0.20   19.3932
9992  Standard Class   Consumer  United States  ...        4     0.00   13.3200
9993    Second Class   Consumer  United States  ...        2     0.00   72.9480

[9994 rows x 13 columns]

Displaying first 5 rows

In [3]:
data.head()

Ship Mode    Segment        Country  ... Quantity Discount    Profit
0    Second Class   Consumer  United States  ...        2     0.00   41.9136
1    Second Class   Consumer  United States  ...        3     0.00  219.5820
2    Second Class  Corporate  United States  ...        2     0.00    6.8714
3  Standard Class   Consumer  United States  ...        5     0.45 -383.0310
4  Standard Class   Consumer  United States  ...        2     0.20    2.5164

[5 rows x 13 columns]

Displaying last 5 rows

In [4]:
data.tail()

Ship Mode   Segment        Country  ... Quantity Discount   Profit
9989    Second Class  Consumer  United States  ...        3      0.2   4.1028
9990  Standard Class  Consumer  United States  ...        2      0.0  15.6332
9991  Standard Class  Consumer  United States  ...        2      0.2  19.3932
9992  Standard Class  Consumer  United States  ...        4      0.0  13.3200
9993    Second Class  Consumer  United States  ...        2      0.0  72.9480

[5 rows x 13 columns]

Checks the number of rows and columns in the dataset

In [5]:
data.shape

(9994, 13)

In [6]:
#Information of the data
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Ship Mode     9994 non-null   object 
 1   Segment       9994 non-null   object 
 2   Country       9994 non-null   object 
 3   City          9994 non-null   object 
 4   State         9994 non-null   object 
 5   Postal Code   9994 non-null   int64  
 6   Region        9994 non-null   object 
 7   Category      9994 non-null   object 
 8   Sub-Category  9994 non-null   object 
 9   Sales         9994 non-null   float64
 10  Quantity      9994 non-null   int64  
 11  Discount      9994 non-null   float64
 12  Profit        9994 non-null   float64
dtypes: float64(3), int64(2), object(8)
memory usage: 1015.1+ KB


Statistical Information

In [7]:
data.describe()

Postal Code         Sales     Quantity     Discount       Profit
count   9994.000000   9994.000000  9994.000000  9994.000000  9994.000000
mean   55190.379428    229.858001     3.789574     0.156203    28.656896
std    32063.693350    623.245101     2.225110     0.206452   234.260108
min     1040.000000      0.444000     1.000000     0.000000 -6599.978000
25%    23223.000000     17.280000     2.000000     0.000000     1.728750
50%    56430.500000     54.490000     3.000000     0.200000     8.666500
75%    90008.000000    209.940000     5.000000     0.200000    29.364000
max    99301.000000  22638.480000    14.000000     0.800000  8399.976000

Checking for the NAN values

In [8]:
data.isna().sum()

Ship Mode       0
Segment         0
Country         0
City            0
State           0
Postal Code     0
Region          0
Category        0
Sub-Category    0
Sales           0
Quantity        0
Discount        0
Profit          0
dtype: int64

No null values

Checking the dataset whether it has duplicated values or no

In [9]:
data.duplicated().sum()

17

There are 17 duplicate values in the dataset

Dropping the duplicate values

In [10]:
data.drop_duplicates(inplace= True)
print("Dropped Duplicate Values")

Dropped Duplicate Values


In [11]:
data.shape

(9977, 13)

before dropping the duplicated values number of rows were 9994 now it is 9977

Deletion of unwanted columns

In [12]:
col=['Postal Code']
data=data.drop(columns=col,axis=1)

Correlation of dataset

In [ ]:
data.corr()

Covariance of dataset

In [ ]:
data.cov()

###Data Visualization

We Will Analyze Segment of people buying from Superstore

In [ ]:
data['Segment'].value_counts()

In [ ]:
segment=(data['Segment'].value_counts()/len(data['Segment'])*100).plot(kind='bar', color='R')
segment

plt.figure(figsize=(6,6))
plt.title('Analyzing Segment of people buying from Superstore')
plt.pie(data['Segment'].value_counts(), labels=data['Segment'].value_counts().index, autopct= '%1.1f%%')
plt.show()


From the graph we come to the conclusion that around 51.9% are the consumers buying from the superstore, 30.2% are the corporate people and 17.8% are the home office people buying from the superstore

We Will Analyze Ship Mode for Superstore

In [ ]:
data['Ship Mode'].value_counts()

In [ ]:
ship_mode=(data['Ship Mode'].value_counts()/len(data['Ship Mode'])*100).plot(kind='bar', color='R')
ship_mode

plt.figure(figsize=(6,6))
plt.title('Analyzing Ship Mode for Superstore')
plt.pie(data['Ship Mode'].value_counts(), labels=data['Ship Mode'].value_counts().index, autopct= '%1.1f%%')
plt.show()


From the above graph most of the deliveries belong to the standard class which is 59.7%, 19.5% belongs to second class, around 15.4% belong to first class and very less deliveries were made on the same day that is 5.4%

Analyzing Sub-Category of items in the superstore

In [ ]:
sub_category=(data['Sub-Category'].value_counts()/len(data['Sub-Category'])*100).plot(kind='bar', color='R')
sub_category

plt.figure(figsize=(6,6))
plt.title('Analyzing Sub-Category of items in Superstore')
plt.pie(data['Sub-Category'].value_counts(), labels=data['Sub-Category'].value_counts().index, autopct= '%1.1f%%')
plt.show()

From the above graph we conclude that :

*    9-15.5% Sub-Category falls under Office Supplies.
*    2-8% Sub-Category belongs to technology and furniture



Analyzing Category, the item that belongs to

In [ ]:
category=(data['Category'].value_counts()/len(data['Category'])*100).plot(kind='bar', color='R')
category

plt.figure(figsize=(6,6))
plt.title('Analyzing Category the items belongs in Superstore')
plt.pie(data['Category'].value_counts(), labels=data['Category'].value_counts().index, autopct= '%1.1f%%')
plt.show()

From the above graph we conclude that Office Supplies has the highest sales which is 60.3% compared to furniture(21.2%) and Technology(18.5%)

Using countplot to find out the maximum orders of the sub-category item

In [ ]:
plt.figure(figsize=(20,10))
sns.countplot(x='Sub-Category',hue="Region", data=data)
plt.xlabel('Sub-Category', size=16 , color = 'Red')
plt.ylabel('No. of Orders', size = 16 ,color = 'Red')
plt.xticks(rotation="vertical", size=20)
plt.show()

From the above graph we conclude that Binders has maximum reach where binders in west region maximum number of orders with respect to other items

Using countplot to plot state and number of orders with respect to region


In [ ]:
plt.figure(figsize=(20,10))
sns.countplot(x='State',hue="Region", data=data)
plt.xlabel('State', size=16 , color = 'Red')
plt.ylabel('No. of Orders', size = 16 ,color = 'Red')
plt.xticks(rotation="vertical", size=20)
plt.show()

From the above observations in the graph we notice that California in west has maximum number of orders and then comes new york east has maximum number of orders

Using bargraph to plot the state that has highest order without region

In [ ]:
plt.figure(figsize=(20,10))
data['State'].value_counts().plot.bar(color = 'red')
plt.xlabel('States', size=16 , color = 'Green')
plt.xticks(size = 20)
plt.ylabel('No. of Orders', size = 16 ,color = 'Green')
plt.show()

From the above plot we observe that california has maximum number of orders and then comes new york

Finding the maximum profit of a particular Item

In [ ]:
profit=pd.DataFrame(data.groupby('Sub-Category').sum()['Profit'])
profit

In [ ]:
profit.plot.bar(figsize=(20,10), grid=True)
plt.show()

From the above graph we notice that profit of Copiers is more compared to other items

Finding the maximum sales of a particular item

In [ ]:
sales=pd.DataFrame(data.groupby('Sub-Category').sum()['Sales'])
sales

In [ ]:
sales.plot.bar(figsize=(20,10), grid=True)
plt.show()

From the above graph we notice that Phones have highest sales then comes chairs

Finding the maximum profit of a particular item that belongs to

In [ ]:
category_profit=pd.DataFrame(data.groupby('Category').sum()['Profit'])
category_profit

In [ ]:
plt.figure(figsize=(20,10))
category_profit.groupby('Category').sum()['Profit'].plot.pie(autopct='%1.1f%%')

From the above graph technology has highest profit which is 50.8%


Finding the maximum sales of a particular item that belongs to

In [ ]:
category_sales=pd.DataFrame(data.groupby('Category').sum()['Sales'])
category_sales

In [ ]:
plt.figure(figsize=(20,10))
category_sales.groupby('Category').sum()['Sales'].plot.pie(autopct='%1.1f%%')

From the above graph technology has highest sales which is 36.4%

Finding out the state that has more profit

In [ ]:
states_profit=pd.DataFrame(data.groupby('State').sum()['Profit'])
states_profit

In [ ]:
states_profit.plot.bar(figsize=(20,10), grid=True)
plt.show()

As shown earlier from the above graph California has highest profit then comes New York

Finding out the state that has more sales

In [ ]:
states_sales=pd.DataFrame(data.groupby('State').sum()['Sales'])
states_sales

In [ ]:
states_sales.plot.bar(figsize=(20,10), grid=True)
plt.show()

From the above graph California has highest sales then comes New York

Finding out Region that has highest sales

In [ ]:
region_sales=pd.DataFrame(data.groupby('Region').sum()['Sales'])
region_sales

In [ ]:
plt.figure(figsize=(20,10))
data.groupby('Region').sum()['Sales'].plot.pie(autopct='%1.1f%%')

From the above pie chart we notice that west region highest sales which 31.6%, east with 29.5% sales, Central has 21.8% and south region has least sales which is 17.1% sales

Finding out Region that has highest profit

In [ ]:
region_profit=pd.DataFrame(data.groupby('Region').sum()['Profit'])
region_profit

In [ ]:
plt.figure(figsize=(20,10))
data.groupby('Region').sum()['Profit'].plot.pie(autopct='%1.1f%%')

From the above pie chart we notice that west region highest profit which 37.8%, east with 32.0% profit ,south has 16.3% profit and Central has least profit which is 13.9%

Summary

   * Subcategory :
           1.  highest profit - Tables
           2.  least profit - Copiers
 
           1.  highest sales - Phones
           2.  least sales - Fasteners 

   * Category :

           1.  highest profit - TECHNOLOGY(50.8%)
           2.  least profit - FURNITURE(6.4%)
 
           1.  highest sales - TECHNOLOGY(36.4%)
           2.  least sales -  OFFICE SUPPLY(31.3%)

   *  State :

           1. highest profit - CALIFORNIA
           2. least profit - TEXAS
 
           1. highest sales - CALIFORNIA
           2. least sales - NORTH DAKOTA

   * Region :

           1. highest profit - WEST (37.8%)
           2. least profit - CENTRAL (13.9%)
 
           1. highest sales - WEST (31.6%)
           2. least sales -  SOUTH (17.1%)